x-pack/plugins/security_solution/common/config_settings.ts

```diff
@@ -22,7 +18,6 @@ export interface ConfigSettings {
  */
 export const defaultSettings: ConfigSettings = Object.freeze({
   ILMEnabled: true,
-  ESQLEnabled: true,
```

Comment:
Remember to remove it from the serverless config

https://github.com/elastic/kibana/blob/60c6cdb9985570a6f58bbf3860539de64ace9aa6/config/serverless.security.yml#L26